### PR TITLE
[Archiver] feat: Unpacker class implementation started

### PR DESCRIPTION
**Pull request by TheCoolKuid**

### CODE REVIEW

File: patcher/archiver.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#include "archiver.h"
+#include "../archiver/archiver.h"
+
+namespace Unpacker
+{
+    Unpacker::Unpacker(const std::filesystem::path &user_path) : user_path{user_path}
+    {
+    }
+
+    std::vector<UnCompressedDataDefinition> Unpacker::Unpack(const std::vector<archiver::archiver::CompressedDataDefinition> &packedFilesHandler)
```
Comment (TheCoolKuid) [inline patcher/archiver.cpp]: unpack info stored as array of FileMemoryDefinition_t, means FileMemoryDefinition_t* not a vector

File: patcher/archiver.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+        size_t files_count{packedFilesHandler.size()};
+
+        /// сделай как надо!!!
+        char *offset = (char *)const_cast<uint8_t *>(data);
+        for (size_t i = 0; i < files_count; i++)
+        {
+            UnCompressedDataDefinition Data;
+
+            size_t CompressedSize{packedFilesHandler[i].CompressedSize};
```
Comment (TheCoolKuid) [inline patcher/archiver.cpp]: rewrite it according FileMemoryDefinition_t struct

File: patcher/archiver.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            size_t OriginSize{packedFilesHandler[i].OriginSize};
+
+            Data.data =
```
Comment (TheCoolKuid) [inline patcher/archiver.cpp]: i suppose move() is required

File: patcher/archiver.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                archiver::archiver::decompress(offset, CompressedSize, OriginSize);
+            offset += CompressedSize;
+            Data.crc32 = archiver::archiver::crc32b(Data.data.get(), OriginSize);
+            Data.OriginSize = OriginSize;
+
+            if (!packedFilesHandler[i].crc32 == Data.crc32)
+            {
+                Data.IsCrcGood = false;
+                throw std::runtime_error("File crc32 value does not match original value!");
```
Comment (TheCoolKuid) [inline patcher/archiver.cpp]: throw will immediatly return from this function so your Data.IsCrcGood  are useless

File: patcher/archiver.cpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            }
+            else
+                Data.IsCrcGood = true;
+
+            out.push_back(std::move(Data));
+        }
```
Comment (TheCoolKuid) [inline patcher/archiver.cpp:81]: where is return?

File: patcher/archiver.h [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    public:
+        /**
+         * @brief Structure describing uncompressed file and success of uncompress
+         *
+         */
+        struct UnCompressedDataDefinition
+        {
+            size_t OriginSize = 0;        ///< uncompressed size
+            std::unique_ptr<char[]> data; ///< uncompressed data
+            unsigned int crc32;           ///< crc32 value (after decompression)
```
Comment (TheCoolKuid) [inline patcher/archiver.h:26]: It unnecessarily to store both crc32 and IsCrcGood. Besides what are you suppose to do with broken data anyway?

**Review by ArturSalamatin — APPROVED** [review]

**Review by ArturSalamatin — APPROVED** [review]

**Pull request merged**